include/rsl/validators.hpp

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <rsl/algorithm.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tl_expected/expected.hpp>
+
+#include <fmt/ranges.h>
+#include <string>
+#include <vector>
+
+namespace rsl {
+
+/** @file */
+
+/**
+ * @brief TODO :(
+ */
+template <typename T>
+[[nodiscard]] auto unique(rclcpp::Parameter const& parameter) -> tl::expected<void, std::string> {
+    if (is_unique(parameter.get_value<std::vector<T>>())) return {};
+    return tl::make_unexpected(
+        fmt::format("Parameter '{}' must only contain unique values", parameter.get_name()));
+}
+
+/**
+ * @brief TODO :(
+ */
+template <typename T>
+[[nodiscard]] auto subset_of(rclcpp::Parameter const& parameter, std::vector<T> const& valid_values)
```

Comment:
Do we want to keep using the `is_` prefix so that this function reads more like plain English? If so, there are many other functions in this file which may also need that prefix.